src/lib/mina_base/party.ml

```diff
@@ -32,17 +32,113 @@ module Update = struct
     [%%versioned
     module Stable = struct
       module V1 = struct
-        type ('state_element, 'pk, 'vk, 'perms, 'snapp_uri, 'token_symbol) t =
+        type ( 'state_element
+             , 'pk
+             , 'vk
+             , 'perms
+             , 'snapp_uri
+             , 'token_symbol
+             , 'timing )
+             t =
           { app_state : 'state_element Snapp_state.V.Stable.V1.t
           ; delegate : 'pk
           ; verification_key : 'vk
           ; permissions : 'perms
           ; snapp_uri : 'snapp_uri
           ; token_symbol : 'token_symbol
+          ; timing : 'timing
+          }
+        [@@deriving compare, equal, sexp, hash, yojson, hlist]
+      end
+    end]
+  end
+
+  module Timing_info = struct
+    [%%versioned
+    module Stable = struct
```

Comment:
Can `Account_timing.t` not be used here?

Reply:
The record there is wrapped in the `Timed` variant, and ends up using a different hash layout etc. than here. I'm not sure it's worth deduplicating just to avoid writing out a similar-but-different type.

Reply:
Ok. Could you add a unit test for creating timed accounts via snapps?